From ceacb67bfaf0ff841ee68b178fe7d9565d2ec2c6 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Mon, 8 Jun 2020 06:54:56 +0530
Subject: [PATCH 23/24] TODO: gio/gcontenttype: Needs porting to WinRT model

Need to think about how content types map to the WinRT world.
---
 gio/gcontenttype-win32.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/gio/gcontenttype-win32.c b/gio/gcontenttype-win32.c
index 11ea203ab..4e1ce5114 100644
--- a/gio/gcontenttype-win32.c
+++ b/gio/gcontenttype-win32.c
@@ -38,6 +38,9 @@ static char *
 get_registry_classes_key (const char    *subdir,
                           const wchar_t *key_name)
 {
+#ifdef G_WINAPI_ONLY_APP
+  return NULL;
+#else
   wchar_t *wc_key;
   HKEY reg_key = NULL;
   DWORD key_type;
@@ -45,7 +48,7 @@ get_registry_classes_key (const char    *subdir,
   char *value_utf8;
 
   value_utf8 = NULL;
-  
+
   nbytes = 0;
   wc_key = g_utf8_to_utf16 (subdir, -1, NULL, NULL, NULL);
   if (RegOpenKeyExW (HKEY_CLASSES_ROOT, wc_key, 0,
@@ -83,6 +86,7 @@ get_registry_classes_key (const char    *subdir,
     RegCloseKey (reg_key);
 
   return value_utf8;
+#endif
 }
 
 /*< private >*/
@@ -403,6 +407,9 @@ g_content_type_guess (const gchar  *filename,
 GList *
 g_content_types_get_registered (void)
 {
+#ifdef G_WINAPI_ONLY_APP
+  return NULL;
+#else
   DWORD index;
   wchar_t keyname[256];
   DWORD key_len;
@@ -434,6 +441,7 @@ g_content_types_get_registered (void)
     }
   
   return g_list_reverse (types);
+#endif
 }
 
 gchar **
-- 
2.44.0.windows.1

